### ISSUE TITLE
Integration Failing?

### ISSUE THREAD

**Issue by scottlitch** [smockle/homebridge-lutron-caseta#27]
I'm considering making the investment to buy a Smart Bridge Pro to run this in HomeBridge and pick up some additional Picos. Can someone give me the likelihood of this integration failing to work in the future? I'm new to this and can barely read case. Am I in over my head?

**Comment by jmissig** [comment]
The maintainer of this plugin is no longer maintaining it. I think it’s best to pressure Lutron for official support.

> I don’t use Lutron Caséta switches or remotes anymore, and I’m unlikely to undertake any new development on this plugin in the near future. For now, new patch versions will continue to be published automatically when this plugin’s depedencies are updated. Please open an issue if you are interested in maintaining this plugin.

**Comment by scottlitch** [comment]
> The maintainer of this plugin is no longer maintaining it. I think it’s best to pressure Lutron for official support.
> 
> > I don’t use Lutron Caséta switches or remotes anymore, and I’m unlikely to undertake any new development on this plugin in the near future. For now, new patch versions will continue to be published automatically when this plugin’s depedencies are updated. Please open an issue if you are interested in maintaining this plugin.

How do we do this? I've already spoke to them on the phone, tweeted and emailed about this.

**Comment by jmissig** [comment]
Wish I knew!

To answer your original question: There are three major pieces being connected by this software. Caséta, HomeKit, and Homebridge.

- Lutron obviously does not move quickly at all, and it seems to me unlikely they’d introduce a major change that would break this anytime soon. Further, as far as I know, this plugin is accessing Caséta via mechanisms intended for third party integration, so Lutron likely wouldn’t want to break that part.
- HomeKit seems to mostly get additions, not major revisions that break all existing clients. Hopefully changes wouldn’t break this soon.
- Homebridge has not really deprecated plugins. It seems to support most of the old styles of plugins it’s gone through. It also seems to just add. Changes seem to come largely from all its dependencies, which get regular updates for security and other reasons.

So from that perspective, if you get it set up and running within your home, and you’re not particularly concerned with keeping all the underlying pieces of software up-to-date, it will probably keep working for some time. We just have to hope there are no severe security issues with, say, Homebridge, that would necessitate changes to this plugin that we cannot figure out.

**Comment by smockle** [comment]
These are good points; I don’t have anything to add. Thanks all for the thoughtful discussion! 👍

[Homebrige v1.0.0](https://github.com/homebridge/homebridge/releases) was released this week. I attempted to address the breaking changes and published a new major version ([`3.x`](https://www.npmjs.com/package/homebridge-lutron-caseta-smockle)) of this plugin. I hope it helps someone, but I haven’t been able to test it personally. The `2.x` releases of this plugin should continue to work with older (pre-`1.x`) versions of Homebridge.
